File: Serenity.Script.UI/FilterPanel/FilteringTypes/DateFiltering.cs
using jQueryApi;
using Serenity.Data;
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;

namespace Serenity
{
    [Imported(ObeysTypeSystem = true)]
    public class DateFiltering : BaseEditorFiltering
    {
        public DateFiltering()
            : base(typeof(DateEditor))
        {
        }

        public override List<FilterOperator> GetOperators()
        {
            return AppendNullableOperators(
                AppendComparisonOperators(
                    new List<FilterOperator>()));
        }
    }
}

File: packages/corelib/src/ui/filtering/datefiltering.ts
import { nsSerenity } from "../../base";
import { DateEditor } from "../editors/dateeditor";
import { BaseEditorFiltering } from "./baseeditorfiltering";
import { FilterOperator } from "./filteroperator";

/**
 * Filtering handler for date fields using a date editor.
 */
export class DateFiltering extends BaseEditorFiltering<DateEditor> {
    static override[Symbol.typeInfo] = this.registerClass(nsSerenity);

    /**
     * Creates a date filtering handler.
     */
    constructor() {
        super(DateEditor)
    }

    /**
     * Returns the operators supported by this filtering handler.
     * @returns The operators.
     */
    getOperators(): FilterOperator[] {
        return this.appendNullableOperators(this.appendComparisonOperators([]));
    }
}
